### PR TITLE
[CHK-1434] feat(npg): add notification callback API

### PR DESCRIPTION
**Pull request by giovanniberti**
#### List of Changes

* feat(npg): add notification callback API

#### Motivation and Context

Added a notification endpoint for Nexi's _Nuovo Payment Gateway_ to be called after the HPP process is completed (see step 17 [here](https://developer.nexigroup.com/it/modalita-di-integrazione/hosted-payment-page)).

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: npg-api/npg-notification-api.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      description: Update Wallet on NPG payment instrument creation
+      operationId: notifyWallet
+      parameters:
+        - in: header
+          name: Correlation-Id
+          schema:
+            type: string
+            format: uuid
+          required: true
+          description: Unique identifier of the the request
```
Comment (ciuffagianluca) [inline npg-api/npg-notification-api.yaml:24]: type: double "the"

Comment (giovanniberti) [reply npg-api/npg-notification-api.yaml]: Fixed in [db82361](https://github.com/pagopa/pagopa-wallet-service/pull/18/commits/db82361accc26a17b5b66337457827422c0aef1f)

**Comment by ciuffagianluca** [comment]
Please add e brief description

**Review by ciuffagianluca — DISMISSED** [review]
LGTM!

File: npg-api/npg-notification-api.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          description: Unique identifier of the request
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/NotificationRequest'
+        required: true
+      responses:
+        '204':
+          description: Payment request sent successfully
```
Comment (infantesimone) [inline npg-api/npg-notification-api.yaml:33]: typo?

**Review by infantesimone — CHANGES_REQUESTED** [review]
same consideration in https://github.com/pagopa/pagopa-infra/pull/902/files

File: npg-api/npg-notification-api.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      responses:
+        '204':
+          description: Notification handled successfully
+        '400':
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemJson'
+        '401':
```
Comment (pietro-tota) [inline npg-api/npg-notification-api.yaml:40]: This status code is to be returned in case of mismatch between received security token and saved one?

**Review by pietro-tota — APPROVED** [review]
LGMT

**Review by infantesimone — APPROVED** [review]

**Review by ciuffagianluca — APPROVED** [review]

File: npg-api/npg-notification-api.yaml (hunk 3 above)
Comment (giovanniberti) [reply npg-api/npg-notification-api.yaml:40]: Exactly :)

**Pull request merged**